fpm/src/fpm.f90

```diff
@@ -126,93 +144,337 @@ subroutine cmd_build(settings)
+       &'  use '//bname//', only: say_hello',    &
+       &'',                                      &
+       &'  implicit none',                       &
+       &'',                                      &
+       &'  call say_hello',                      &
+       &'end program main']
+      call warnwrite(join_path(settings%name, 'app/main.f90'), littlefile)
    endif
 
-   write(*,'(a)')message
-   print *, "fpm error: 'fpm new' not implemented."
-   error stop 1
-end subroutine cmd_new
+   call warnwrite(join_path(settings%name, 'fpm.toml'), message)               ! now that built it write NAME/fpm.toml
 
-subroutine cmd_run(settings)
-    type(fpm_run_settings), intent(in) :: settings
-    integer                      :: i
-
-    write(*,*)'RELEASE=',settings%release
-    if(size(settings%name).eq.0)then
-        write(*,*)'RUN DEFAULTS with arguments ['//settings%args//']'
-    else
-        do i=1,size(settings%name)
-            write(*,*)'RUN:'//trim(settings%name(i))//' with arguments ['//settings%args//']'
-        enddo
-    endif
-
-    print *, "fpm error: 'fpm run' not implemented."
-    error stop 1
+   call run('cd ' // settings%name // ';git init')    ! assumes these commands work on all systems and git(1) is installed 
+contains
+!===================================================================================================================================
+subroutine warnwrite(fname,data)
+character(len=*),intent(in) :: fname
+character(len=*),intent(in) :: data(:)
+   if(.not.exists(fname))then
+      call filewrite(fname,data)
+   else
+      write(stderr,'(*(g0,1x))')'fpm::new<WARNING>',fname,'already exists. Not overwriting'
+   endif
+end subroutine warnwrite
+!===================================================================================================================================
+subroutine filewrite(filename,filedata)
+use,intrinsic :: iso_fortran_env, only : stdin=>input_unit, stdout=>output_unit, stderr=>error_unit
+! write filedata to file filename
+character(len=*),intent(in)           :: filename
+character(len=*),intent(in)           :: filedata(:)
+integer                               :: lun, i, ios
+character(len=256)                    :: message
+   message=' '
+   ios=0
+   if(filename.ne.' ')then
+    open(file=filename, &
+    & newunit=lun, &
+    & form='formatted', &      !  FORM      =  FORMATTED   |  UNFORMATTED
+    & access='sequential', &   !  ACCESS    =  SEQUENTIAL  |  DIRECT       |  STREAM
+    & action='write', &        !  ACTION    =  READ|WRITE  |  READWRITE
+    & position='rewind', &     !  POSITION  =  ASIS        |  REWIND       |  APPEND
+    & status='new', &          !  STATUS    =  NEW         |  REPLACE      |  OLD     |  SCRATCH   | UNKNOWN
+    & iostat=ios, &
+    & iomsg=message)
+   else
+      lun=stdout
+      ios=0
+   endif
+   if(ios.ne.0)then
+      write(stderr,'(*(a,1x))')'*filewrite* error:',filename,trim(message)
```

Comment:
Shouldn't this be either `'(*(a:,1x))'` or `'(*(1x,a))` to avoid trailing whitespace in the output?

Reply:
Technically X sets the position for the next output, like T and '(\*(a,1x))' s not equivalent to '(\*(a," "))' and should not product a trailing space but I have seen lots of differences or errors between compilers like this and it is confusing. gfortran does the "right" thing and leaves no trailing space but does put spaces between the non-advancing I/O in the
sample below but I think your suggestion to make it explicit and to use '(*(a:,1x))' is a good one. Most people either expect the non-advancing I/O to not have spaces or for the output to have trailing spaces in the following:
```fortran
program testit
do i=1,10
   write(*,'(a,1x)',advance='no')'X'
enddo
do i=1,10
   write(*,'(a,1x)')'X'
enddo
end program testit
```
So I like that change. Especially since I was involved in the war over whether a T was relative to the beginning of the line or the output of the current I/O statement when non-advancing I/O was introduced. Several early adopters had already written it to use a buffer an be relative to the entire current line and were wont to change it to what was decided when that was clarified.

Reply:
Thanks for the clarification.